main/lsp/request_dispatch.cc

```diff
@@ -312,165 +23,52 @@ void LSPLoop::processRequest(std::unique_ptr<LSPMessage> msg) {
 }
 
 void LSPLoop::processRequests(vector<unique_ptr<LSPMessage>> messages) {
-    QueueState state;
-    absl::Mutex mutex;
     for (auto &message : messages) {
-        preprocessor.preprocessAndEnqueue(state, move(message), mutex);
+        preprocessor.preprocessAndEnqueue(move(message));
     }
-    ENFORCE(state.paused == false, "__PAUSE__ not supported in single-threaded mode.");
-    for (auto &message : state.pendingRequests) {
-        processRequestInternal(*message);
+    {
+        absl::MutexLock lck(taskQueueMutex.get());
+        ENFORCE(taskQueue->paused == false, "__PAUSE__ not supported in single-threaded mode.");
+        for (auto &task : taskQueue->pendingTasks) {
+            runTask(move(task));
+        }
+        taskQueue->pendingTasks.clear();
     }
 }
 
-void LSPLoop::processRequestInternal(LSPMessage &msg) {
-    // Note: Before this function runs, LSPPreprocessor has already early-rejected any invalid messages sent prior to
-    // the initialization handshake. So, we know that `msg` is valid to process given the current state of the server.
-    auto &logger = config->logger;
-    // TODO(jvilk): Make Timer accept multiple FlowIds so we can show merged messages correctly.
```

Comment:
A bunch of the red in here has moved into LSPPreprocessor, into the method that converts an LSPMessage into an LSPTask.